src/limb.rs

```diff
@@ -48,133 +46,23 @@ pub enum LimbMask {
-    //
-    // TODO: DRY-up with `ec::suite_b::private_key::generate_private_key`.
-    fn sample_into_limbs(&self, dest: &mut [Limb], rng: &rand::SecureRandom)
-                         -> Result<(), error::Unspecified> {
-        // Loosely based on [NSA Suite B Implementer's Guide to ECDSA]
-        // Appendix A.1.2, and
-        // [NSA Suite B Implementer's Guide to NIST SP 800-56A] Appendix B.2,
-        // "Key Pair Generation by Testing Candidates".
-        //
-        // [NSA Suite B Implementer's Guide to ECDSA]: doc/ecdsa.pdf.
-        // [NSA Suite B Implementer's Guide to NIST SP 800-56A]: doc/ecdh.pdf.
-
-        assert_eq!(self.max_exclusive.len(), dest.len());
-
-        let most_significant_limb =
-            self.max_exclusive[self.max_exclusive.len() - 1];
-        debug_assert!(most_significant_limb > 0);
-        let most_significant_limb_mask =
-            most_significant_limb_mask_variable_time(most_significant_limb);
-        debug_assert!(most_significant_limb_mask != 0);
-
-        // XXX: The value 100 was chosen to match OpenSSL due to uncertainty of
-        // what specific value would be better, but it seems bad to try 100
-        // times.
-        for _ in 0..100 {
-            {
-                let mut dest_as_bytes = limbs_as_bytes_mut(dest);
-                try!(rng.fill(&mut dest_as_bytes));
-            }
-            dest[self.max_exclusive.len() - 1] &= most_significant_limb_mask;
-
-            if self.are_limbs_within(&dest) {
-                return Ok(());
-            }
-        }
-
-        Err(error::Unspecified)
-    }
-}
-
-/// Returns a mask that has the same number of leading zeros as
-/// `most_significant_limb`, with all the following bits set.
-fn most_significant_limb_mask_variable_time(most_significant_limb: Limb)
-                                            -> Limb {
-    const ONE: Limb = 1;
-
-    let bits_to_mask_off = most_significant_limb.leading_zeros();
-    if bits_to_mask_off == 0 {
-        Limb::max_value()
-    } else {
-        (ONE << (LIMB_BITS_U32 - bits_to_mask_off)) - 1
-    }
-}
-
-#[allow(unsafe_code)]
-#[allow(non_snake_case)]
-#[doc(hidden)]
-#[no_mangle]
-pub unsafe extern fn GFp_rand_mod(dest: *mut Limb, max_exclusive: *const Limb,
-                                  num_limbs: c::size_t, rng: *mut rand::RAND)
-                                  -> c::int {
-    const ERR: c::int = 0;
-    const SUCCESS: c::int = 1;
-
-    let range = Range::from_max_exclusive(
-        core::slice::from_raw_parts(max_exclusive, num_limbs));
-    let mut dest = core::slice::from_raw_parts_mut(dest, num_limbs);
-
-    let result = range.sample_into_limbs(&mut dest, (*rng).rng);
-    if result.is_err() {
-        return ERR;
-    }
-
-    SUCCESS
-}
-
-
 #[cfg(all(test, target_pointer_width = "64"))]
-fn limbs_as_bytes<'a>(src: &'a [Limb]) -> &'a [u8] {
+pub fn limbs_as_bytes<'a>(src: &'a [Limb]) -> &'a [u8] {
```

Comment:
I suspect these small functions should be annotated `#[inline]` to make sure they get inlined cross-module.